R/expr__expr.R

```diff
@@ -681,6 +687,9 @@ Expr_map = function(f, output_type = NULL, agg_list = FALSE, in_background = FAL
 #'  investigate.
 #' @name Expr_apply_in_background
 Expr_apply_in_background = function(f, output_type = NULL) {
+  if (Sys.getenv("RPOLARS_R_BINARY_PATH") == "") {
```

Comment:
[envvars are not thread safe](https://stackoverflow.com/questions/30476600/how-can-getenv-be-thread-safe). We would probably be fine in most cases though.

I have read py-polars have similar envvar where it is discouraged to modify it after start-up (.onLoad()). If we really needed a thread safe abstraction we could use the `Storage`-crate as with the global [like here in `lib.rs`](https://github.com/pola-rs/r-polars/blob/5d4a59bd6e05f59ffc1ad9759a0c022caeffcdb9/src/rust/src/lib.rs#L36).

I think we might be a bit better of moving this Default setting to [.onLoad](https://github.com/pola-rs/r-polars/blob/5d4a59bd6e05f59ffc1ad9759a0c022caeffcdb9/R/zzz.R#L119)

We get mod this

Reply:
Why does this needs to be thread safe? From my understanding the only scenario where this function could be called is when users are constructing expressions in the single-threaded R session.

Reply:
Depends on how easy it is to copy the `Static Lazy` pattern. It just in principle cleanest to have as few "invariants" as possible.Some point in the future some careless dev may refactor your code and then it is good to have as few "not causing havoc right now"-elements as possible.

Currently the thread "message container" struct `ParRobj` only works with plain R vectors, not R functions across. There is some type checking, but maybe some altRep, might make it blow in the future. The function  `map_alias()` do contain invariant it should be run from the main thread. I just missed some PyO3 features to implement this cleanly thread-safe. I had an experimental warning for a long time in docs.

So if not worth I also suggest to move the code pieces  to `.onLoad.` I imagine this is a R session related setting, not a query specific setting. In the case a user serializes the query to disk and loads it from another machine it would not work. Maybe you have other scenarios in mind?